Final task

eisina opened the pull request.

eisina committed:
Adding Webdriver, LoginPage, LoginTest classes

eisina committed:
Adding logger
Adding methods in Login Page.
Creating BasePage and moving wedriver initialization to this class
Adding test steps to Lofin test

eisina committed:
Pages added AccountCreationPage, AccountPage, ProductPage, WishList Page
Tests are added for Wishlists
Test is added for cart

eisina committed:
Test is added for cart
Methods added to CartPage

eisina committed:
Alignment fixed

eisina committed:
Json parser, json, User class added for storing and retrieving data

eisina committed:
hunit and allure

eisina committed:
Changing Junit to TestNG

eisina committed:
Adding checks to LoginTest

KBugrimCS commented on src/main/coherentsolutions.web.test/java/Managers/WebDriverManager.java:
```diff
@@ -0,0 +1,66 @@
+    public static WebDriver initDriver() throws IOException {
+        if (driver == null) {
+            propertiesUtils = new PropertiesUtils();
+            switch (propertiesUtils.getProperty("environment")) {
+                case "local": {
+                    switch (propertiesUtils.getProperty("browser")) {
+                        case "chrome": {
+                            driver = new ChromeDriver();
+                            driver.manage().window().maximize();
+                            driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+                            break;
+                        }
+                        case "firefox": {
+                            driver = new FirefoxDriver();
+                            driver.manage().window().maximize();
+                            driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+                            break;
+                        }
+                    }
+                }
+                case "remote": {
+                    if (driver == null) {
+                        ChromeOptions options = new ChromeOptions();
+                        MutableCapabilities caps = new MutableCapabilities();
+                        caps.setCapability("goog:chromeOptions", options);
+                        caps.setCapability("browserName", "chrome");
+                        caps.setCapability("browserVersion", "latest");
+                        caps.setCapability("platformName", "Windows 8.1");
+                        URL url = new URL("https://oauth-isinacath378-f829a:eb59a0a6-bb06-4b1c-82ab-2592e883a227@ondemand.us-west-1.saucelabs.com:443/wd/hub");
+
+                        driver = new RemoteWebDriver(url, caps);
+                        driver.manage().window().maximize();
+                        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+                    }
+                }
+            }
+        }
```
too complicated logic. Please use here interface with different implementations

KBugrimCS commented on src/main/coherentsolutions.web.test/java/Utils/TestDataUtils.java:
```diff
@@ -0,0 +1,29 @@
+    private final String userFilePath = "src/main/resources/testDataResources/user.json";
+    private User user;
+
+    public User getUserData() {
+        Gson gson = new Gson();
+        BufferedReader bufferReader = null;
+        try {
+            bufferReader = new BufferedReader(new FileReader(userFilePath));
+            user = gson.fromJson(bufferReader, User.class);
+            return user;
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Json file not found at path : " + userFilePath);
+        } finally {
+            try {
+                if (bufferReader != null) bufferReader.close();
+            } catch (IOException ignore) {
+            }
+        }
+    }
```
Please use Jackson for mapping .json to user object

eisina committed:
Adding WebDriver Inteface
Adding logger to classes
adding methods for login and populating required fields
Removing Constants and adding info to properties

eisina committed:
Adding allure Feature
